src/Hal/Component/OOP/Resolver/TypeResolver.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * (c) Jean-François Lépine <https://twitter.com/Halleck45>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Hal\Component\OOP\Resolver;
+
+
+/**
+ * Type resolver
+ *
+ * @author Jean-François Lépine <https://twitter.com/Halleck45>
+ */
+class TypeResolver
+{
+
+    const TYPE_VOID = 'void';
+    const TYPE_STRING = 'string';
+    const TYPE_INTEGER = 'integer';
+    const TYPE_FLOAT = 'float';
+    const TYPE_ARRAY = 'array';
+    const TYPE_NULL = 'null';
+    const TYPE_UNKNWON = 'unknown';
+    const TYPE_FLUENT_INTERFACE = 'fluent';
+    const TYPE_ANONYMOUS_CLASS = 'anonymous@class';
+
+    /**
+     * Resolves type of given string
+     *
+     * @param $string
+     * @return string
+     */
+    public function resolve($string)
+    {
+        $string = $cased = trim($string, ";\t\n\r\0\x0B");
+        $string = strtolower($string);
+
+        if(strlen($string) == 0) {
+            return self::TYPE_VOID;
+        }
+
+        if(preg_match('!^\d+$!', $string)) {
+            return self::TYPE_INTEGER;
+        }
+
+        if(preg_match('!^\d+\.\d+$!', $string)) {
+            return self::TYPE_FLOAT;
+        }
+
+        if('null' == $string) {
+            return self::TYPE_NULL;
+        }
+
+        if(preg_match('!(^\[|^array\()!', $string)) {
+            return self::TYPE_ARRAY;
+        }
+
+        if(preg_match('!^new\s+class\s+!', $string, $matches)) {
+            return self::TYPE_ANONYMOUS_CLASS;
+        }
+
+        if(preg_match('!^(new\s+)(.*?)([\(;].*|$)!', $cased, $matches)) {
+            return $matches[2];
```

Comment:
@stof that's an idea. But I don't want to get "object". My need is to get the "function" of the given string (if possible). In PhpMetrics, I think (but I'm not sure) that anonymous classes should not be considered as classical classes : they have impact on afferent coupling, but generally they are like "inner class", and their visibility is localized.

In fact, when we consider returned types, only classes or interfaces are used by PhpMetrics in calculations.